x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/role_selector.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { startCase } from 'lodash';
+
+import {
+  EuiCallOut,
+  EuiFormRow,
+  EuiRadio,
+  EuiSpacer,
+  EuiText,
+  EuiTextColor,
+  EuiTitle,
+} from '@elastic/eui';
+
+interface IRoleSelectorProps {
+  disabled?: boolean;
+  disabledText?: string;
+  roleType?: string;
+  roleTypeOption: string;
+  description: string;
+  onChange(roleTypeOption: string): void;
+}
+
+export const RoleSelector: React.FC<IRoleSelectorProps> = ({
+  disabled,
+  disabledText,
+  roleType,
+  roleTypeOption,
+  description,
+  onChange,
+}) => (
+  <EuiFormRow>
+    <EuiRadio
+      disabled={disabled}
+      id={roleTypeOption}
+      checked={roleTypeOption === roleType}
+      onChange={() => {
+        onChange(roleTypeOption);
+      }}
+      label={
+        <>
+          <EuiTitle size="xs">
+            <h4 className="users-layout__users--roletype">{startCase(roleTypeOption)}</h4>
```

Comment:
Do you know if we need this className or not? Kibana classNames should be camelCased, I believe.

```suggestion
            <h4 className="usersLayout__users--roletype">{startCase(roleTypeOption)}</h4>
```

You could also just remove it if you're not sure yet.